src/utils/rtl.tsx

```diff
@@ -180,3 +181,111 @@ function getTextFromTableCellNode(node: ChildNode) {
+    if (!listbox) {
+      _click(select);
+    }
+    const options: NodeListOf<HTMLElement> = body.querySelectorAll("[role=option]");
+    // Allow searching for options by their data-key (value) or textContent (label)
+    const optionToSelect = Array.from(options).find(
+      (o: HTMLElement) => o.dataset.key === optionValue || o.dataset.label === optionValue,
+    );
+    if (!optionToSelect) {
+      throw new Error(`Could not find option with value or text content of ${optionValue}`);
+    }
+    _click(optionToSelect);
+  }
+
+  const optionValues = Array.isArray(value) ? value : [value];
+  optionValues.forEach((optionValue) => {
+    maybeOpenAndSelect(optionValue);
+  });
+}
+
+export function selectAndWait(input: HTMLElement, value: string | string[]): Promise<void> {
+  return allowAndWaitForAsyncBehavior(() => select(input, value));
+}
+
+export function getSelected(element: HTMLElement): string[] | string | undefined {
+  const select = resolveIfNeeded(element);
+  if (isSelectElement(select)) {
+    throw new Error("Beam getSelected helper does not support <select> elements");
+  }
+
+  if (!isInputElement(select) && select.dataset.readonly === "true") {
+    // For read-only fields that render as a 'div'
+    return select.textContent ?? undefined;
+  }
+
+  const listboxId = select.getAttribute("aria-controls");
+  const body = select.closest("body") as HTMLElement;
+  const listbox = body.querySelector(`#${listboxId}`) as HTMLElement;
+  if (!listbox) {
+    _click(select);
+  }
+  const options: NodeListOf<HTMLElement> = body.querySelectorAll("[role=option]");
+
+  const selections: string[] = Array.from(options)
+    .filter((o: HTMLElement) => o.getAttribute("aria-selected") === "true")
+    .map((o: HTMLElement) => o.dataset.label ?? o.dataset.key ?? "")
+    // Filter out empty strings
+    .filter((o) => !!o);
+
+  return selections.length > 0 ? (selections.length > 1 ? selections : selections[0]) : undefined;
+}
+
+export function getOptions(element: HTMLElement): string[] {
+  const select = resolveIfNeeded(element);
+  if (isSelectElement(select)) {
+    throw new Error("Beam getOptions helper does not support <select> elements");
+  }
+
+  if (!isInputElement(select)) {
+    throw new Error(
+      `Expected element to be INPUT, but got ${select.nodeName}. This field may be read-only. In that case we cannot get the list of options`,
+    );
+  }
+
+  const listboxId = select.getAttribute("aria-controls");
+  const body = select.closest("body") as HTMLElement;
+  const listbox = body.querySelector(`#${listboxId}`) as HTMLElement;
+  if (!listbox) {
+    _click(select);
+  }
+  const options: NodeListOf<HTMLElement> = body.querySelectorAll("[role=option]");
+
+  return Array.from(options)
+    .map((o: HTMLElement) => o.dataset.label ?? o.dataset.key ?? "")
+    .filter((o) => !!o);
+}
+
+function resolveIfNeeded(element: HTMLElement): HTMLElement {
```

Comment:
Fwiw once the React 18 bump re-lands (I'm waiting on zgavin to land a backend PDF PR) I'm planning on de-lazy-ing all of our `r.foo`s to do immediate lookups, so we won't need the `r.foo()` anymore. Looking forward to it!